lgc/patch/Patch.cpp

```diff
@@ -378,7 +378,8 @@ void LegacyPatch::addPasses(PipelineState *pipelineState, legacy::PassManager &p
 // @param [in/out] passMgr : Pass manager to add passes to
 // @param optLevel : The optimization level uses to adjust the aggressiveness of
 //                   passes and which passes to add.
-void Patch::addOptimizationPasses(lgc::PassManager &passMgr, CodeGenOpt::Level optLevel) {
+// @param disableScalarizer : Whether the scalarizer pass should be disabled.
+void Patch::addOptimizationPasses(lgc::PassManager &passMgr, CodeGenOpt::Level optLevel, bool disableScalarizer) {
```

Comment:
Boolean arguments tend to be a bad idea if they're not completely obvious. What if somebody wants to do this for another pass?

How about passing the PipelineState instead, just as for Patch::addPasses?